src/main/kotlin/com/zaxxer/ping/IcmpPinger.kt

```diff
@@ -333,11 +354,18 @@ class IcmpPinger(private val responseHandler:PingResponseHandler) {
          cc -= headerLen
 
          icmp.useMemory(socketBufferPointer.slice(headerLen.toLong()))
-         if (icmp.icmp_type.get() != ICMP_ECHOREPLY) {
+         icmp6.useMemory(socketBufferPointer)
+
+         if (icmp.icmp_type.get() != ICMP_ECHOREPLY && icmp6.icmp6_type.get() != ICMPV6_ECHO_REPLY) {
```

Comment:
Isn't this dangerous?  Are the ``icmp.icmp_type`` and ``icmp6.icmp6_type`` fields at the same location in the reply packet?  Is it possible for the ``icmp.icmp_type`` memory location to have a value of ``ICMP_ECHOREPLY`` just by accident in the case of an IPv6 reply?

Reply:
icmp.icmp_type and icmp6.icmp6_type are in the same location of the reply packet.

Reply:
icmp.icmp_type

```
class Icmp:Struct(runtime) {
   // u_char	icmp_type;		/* type of message, see below */
   // u_char	icmp_code;		/* type sub code */
   // u_short	icmpCksum;		/* ones complement cksum of struct */
   val icmp_type = Unsigned8()
   val icmp_code = Unsigned8()
   val icmp_cksum = Unsigned16()
   // union {
   val icmp_hun:Hun = inner(Hun())
   // } icmp_hun;
   // union {
   val icmp_dun:Dun = inner(Dun())
   // } icmp_dun
}
```

icmp6.icmp6_type

```
class Icmp6:Struct(runtime) {
   // u_char	icmp_type;		/* type of message, see below */
   // u_char	icmp_code;		/* type sub code */
   // u_short	icmpCksum;		/* ones complement cksum of struct */
   val icmp6_type = Unsigned8()
   val icmp6_code = Unsigned8()
   val icmp6_cksum = Unsigned16()
   // union {
   val icmp6_dataun:Icmp6UnData = inner(Icmp6UnData())
   // } icmp6_dataun;
}
```